shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -2339,6 +2339,75 @@ void releaseInternal() {
         }
     }
 
+    /**
+     * A generic error response that encapsulates an error code (to be used for client-side processing) and an error message
+     * describing the origin of the error. This should be used to describe general exceptions where limited information is required.
+     */
+    @Data
+    public static final class ErrorMessage implements Reply, WireCommand {
+        final WireCommandType type = WireCommandType.ERROR_MESSAGE;
+        final long requestId;
+        final String message;
+        final ErrorCode errorCode;
+
+        @Override
+        public void process(ReplyProcessor cp) throws UnsupportedOperationException {
+            cp.errorMessage(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeInt(errorCode.getCode());
+            out.writeUTF(message);
+        }
+
+        public static WireCommand readFrom(EnhancedByteBufInputStream in, int length) throws IOException {
+            return new ErrorMessage(in.readLong(), in.readUTF(), ErrorCode.valueOf(in.readInt()));
+        }
+
+        public enum ErrorCode {
+            UNSPECIFIED(-1, null),                                         // indicates un-specified (for backward compatibility
```

Comment:
While this may increase the risk of a `NullPointerException`, `null` appears to be a valid return for unknown causes (see: https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getCause--).